migration/idempotent/001-base.sql

```diff
@@ -270,22 +270,23 @@ CREATE OR REPLACE FUNCTION _prom_catalog.make_metric_table()
     RETURNS trigger
     AS $func$
 DECLARE
-  label_id INT;
-  compressed_hypertable_name text;
+    label_id INT;
+    compressed_hypertable_name text;
 BEGIN
 
-   -- Note: if the inserted metric is a view, nothing to do.
-   IF NEW.is_view THEN
+    -- Note: if the inserted metric is a view, nothing to do.
+    IF NEW.is_view THEN
         RETURN NEW;
-   END IF;
+    END IF;
 
-   EXECUTE format('CREATE TABLE %I.%I(time TIMESTAMPTZ NOT NULL, value DOUBLE PRECISION NOT NULL, series_id BIGINT NOT NULL) WITH (autovacuum_vacuum_threshold = 50000, autovacuum_analyze_threshold = 50000)',
+    EXECUTE format('CREATE TABLE %I.%I(time TIMESTAMPTZ NOT NULL, value DOUBLE PRECISION NOT NULL, series_id BIGINT NOT NULL) WITH (autovacuum_vacuum_threshold = 50000, autovacuum_analyze_threshold = 50000)',
                     NEW.table_schema, NEW.table_name);
-   EXECUTE format('GRANT SELECT ON TABLE %I.%I TO prom_reader', NEW.table_schema, NEW.table_name);
-   EXECUTE format('GRANT SELECT, INSERT ON TABLE %I.%I TO prom_writer', NEW.table_schema, NEW.table_name);
-   EXECUTE format('GRANT SELECT, INSERT, UPDATE, DELETE, TRIGGER ON TABLE %I.%I TO prom_modifier', NEW.table_schema, NEW.table_name);
-   EXECUTE format('CREATE UNIQUE INDEX data_series_id_time_%s ON %I.%I (series_id, time) INCLUDE (value)',
+    EXECUTE format('GRANT SELECT ON TABLE %I.%I TO prom_reader', NEW.table_schema, NEW.table_name);
+    EXECUTE format('GRANT SELECT, INSERT ON TABLE %I.%I TO prom_writer', NEW.table_schema, NEW.table_name);
+    EXECUTE format('GRANT SELECT, INSERT, UPDATE, DELETE, TRIGGER ON TABLE %I.%I TO prom_modifier', NEW.table_schema, NEW.table_name);
+    EXECUTE format('CREATE UNIQUE INDEX data_series_id_time_%s ON %I.%I (series_id, time) INCLUDE (value)',
                     NEW.id, NEW.table_schema, NEW.table_name);
+    EXECUTE format('ALTER EXTENSION promscale ADD TABLE %I.%I', NEW.table_schema, NEW.table_name);
```

Comment:
good point. adding the views and exemplar tables too.

Reply:
```ERROR: pg_extension_config_dump() can only be called from an SQL script executed by CREATE EXTENSION (SQLSTATE 0A000)```

we cannot make dynamically created tables a part of the extension because we cannot also make them config tables, and if they aren’t config tables then their data won’t get dumped

Reply:
Okay. I think that this makes sense generally. Because the extension doesn't own these tables, I expect that they should be dumped when running pg_dump anyway (we really need the pg_dump/pg_restore tests).